bind.sh

```diff
@@ -13,92 +13,110 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
-
-set -euo pipefail
 
-# Usage: bind.sh <launcher> [--cpus <spec>] [--gpus <spec>] [--mems <spec>] [--nics <spec>] <app> ...
+# Usage: bind.sh --launcher <launcher> [--cpus <spec>] [--gpus <spec>] [--mems <spec>] [--nics <spec>] -- <app> ...
 # <spec> specifies the resources to bind each node-local rank to, with ranks
 # separated by /, e.g. 0,1/2,3/4,5/6,7 for 4 ranks per node.
 
-# Detect node-local rank based on launcher
-IDX=none
-case "$1" in
-    mpirun) IDX="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
-    jsrun) IDX="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
-    srun) IDX="$SLURM_LOCALID" ;;
-    local) IDX=0 ;;
-    none) IDX="${SLURM_LOCALID:-${OMPI_COMM_WORLD_LOCAL_RANK:-${MV2_COMM_WORLD_LOCAL_RANK:-none}}}" ;;
+set -euo pipefail
+
+help() {
+  echo "Usage: bind.sh -l | --launcher <mpirun,srun,jrun,local> [ -c | --cpus ] [ -g | --gpus ] [ -m | --mems ] [ -n | --nics ] -- <app>"
+  exit 2
+}
+
+while :
+do
+  case "$1" in
+    --launcher) launcher="$2" ;;
+    --cpus) cpus="$2" ;;
+    --gpus) gpus="$2" ;;
+    --mems) mems="$2" ;;
+    --nics) nics="$2" ;;
+    --help) help ;;
+    --)
+      shift;
+      break
+      ;;
+    *)
+      echo "Unexpected option: $1"
+      help
+      ;;
+  esac
+  shift 2
+done
+
+if [ -z "$launcher" ]; then
+  echo "bind.sh: -l / --launcher is a required argument"
+fi
+
+case "$launcher" in
+  local) rank=0 ;;
+  mpirun) rank="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
+  jsrun) rank="$OMPI_COMM_WORLD_LOCAL_RANK" ;;
+  srun) rank="$SLURM_LOCALID" ;;
+  *)
+    echo "Unexpected launcher value: $launcher"
+    help
+    ;;
 esac
-shift
-if [[ "$IDX" == "none" ]]; then
-    echo "Error: Cannot detect node-local rank" 1>&2
-    exit 1
+
+export LEGATE_RANK="$rank"
```

Comment:
Is this env var name OK, or is there something better?

Reply:
Also: do we want to also ultimately change the `nsys` and `nvprof` invocations to utilize this env var instead of the `%q` special case?

Reply:
The name is good. Yes, we can handle nsys and nvprof in the same script as cprofile, and use the same variable to name their output files.